### PR TITLE
Remove references to xl/fs

### PR DESCRIPTION
**Pull request by nitisht**
## Description
- Add a section on Minio modes in top level readme
- Update Shared-backend readme to talk about distributed fs mode
- Update erasure code readme to talk about standalone erasure mode
- Update distributed mode readme to talk about distributed erasure mode

## Motivation and Context
This is an effort to unify terminology across our docs so that different Minio modes are clearly explained and there are no confusions for the end users. Also see https://github.com/minio/minio/issues/3722

## How Has This Been Tested?
Manually

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docs/erasure/README.md [hunk 1]
```diff
@@ -1,10 +1,18 @@
-# Minio Erasure Code Quickstart Guide [![Slack](https://slack.minio.io/slack?type=svg)](https://slack.minio.io)
+# Minio Standalone Erasure Code Quickstart Guide [![Slack](https://slack.minio.io/slack?type=svg)](https://slack.minio.io)
```
Comment (krishnasrinivas) [inline docs/erasure/README.md]: you can remove "Code" here to be consistent

**Review by harshavardhana — DISMISSED** [review]

File: docs/erasure/README.md (hunk 1 above)
Comment (nitisht) [reply docs/erasure/README.md]: done @krishnasrinivas

**Comment by nitisht** [comment]
As discussed with @abperiasamy we wont be taking this, but add a new document that talks about various storage backends Minio supports

**Comment by nitisht** [comment]
New issue #4529

**Pull request closed without merging**